pkg/bgpv1/manager/reconcilerv2/interface.go

```diff
@@ -0,0 +1,240 @@
+	if err != nil {
+		return err
+	}
+
+	return r.reconcilePaths(ctx, p, desiredPeerAdverts, txn)
+}
+
+func (r *InterfaceReconciler) getDesiredPaths(desiredPeerAdverts PeerAdvertisements, txn statedb.ReadTxn) AFPathsMap {
+	desiredAdverts := make(AFPathsMap)
+	for _, peerFamilyAdverts := range desiredPeerAdverts {
+		for family, familyAdverts := range peerFamilyAdverts {
+			agentFamily := types.ToAgentFamily(family)
+			pathsPerFamily, exists := desiredAdverts[agentFamily]
+			if !exists {
+				pathsPerFamily = make(PathMap)
+				desiredAdverts[agentFamily] = pathsPerFamily
+			}
+			for _, advert := range familyAdverts {
+				for _, prefix := range r.getInterfacePrefixes(advert, agentFamily, txn) {
+					path := types.NewPathForPrefix(prefix)
+					path.Family = agentFamily
+					pathsPerFamily[path.NLRI.String()] = path
+				}
+			}
+		}
+	}
+	return desiredAdverts
+}
+
+func (r *InterfaceReconciler) getDesiredRoutePolicies(desiredPeerAdverts PeerAdvertisements, txn statedb.ReadTxn) (RoutePolicyMap, error) {
+	desiredPolicies := make(RoutePolicyMap)
+	for peer, peerFamilyAdverts := range desiredPeerAdverts {
+		if peer.Address == "" {
+			continue
+		}
+		peerAddr, err := netip.ParseAddr(peer.Address)
+		if err != nil {
+			return nil, fmt.Errorf("failed to parse peer address: %w", err)
+		}
+		for family, familyAdverts := range peerFamilyAdverts {
+			agentFamily := types.ToAgentFamily(family)
+			for _, advert := range familyAdverts {
+				var v4Prefixes, v6Prefixes types.PolicyPrefixMatchList
+				for _, prefix := range r.getInterfacePrefixes(advert, agentFamily, txn) {
+					match := &types.RoutePolicyPrefixMatch{CIDR: prefix, PrefixLenMin: prefix.Bits(), PrefixLenMax: prefix.Bits()}
+					if agentFamily.Afi == types.AfiIPv4 {
+						v4Prefixes = append(v4Prefixes, match)
+					}
+					if agentFamily.Afi == types.AfiIPv6 {
+						v6Prefixes = append(v6Prefixes, match)
+					}
+				}
+				if len(v6Prefixes) > 0 || len(v4Prefixes) > 0 {
+					name := PolicyName(peer.Name, agentFamily.Afi.String(), advert.AdvertisementType, "")
+					policy, err := CreatePolicy(name, peerAddr, v4Prefixes, v6Prefixes, advert)
+					if err != nil {
+						return nil, err
+					}
+					desiredPolicies[name] = policy
+				}
+			}
+		}
+	}
+	return desiredPolicies, nil
+}
+
+func (r *InterfaceReconciler) getInterfacePrefixes(advert v2.BGPAdvertisement, family types.Family, txn statedb.ReadTxn) []netip.Prefix {
+	var res []netip.Prefix
+	if advert.Interface == nil {
+		return nil
+	}
+	dev, _, found := r.deviceTable.Get(txn, tables.DeviceNameIndex.Query(advert.Interface.Name))
+	if !found {
+		return nil
+	}
+	for _, addr := range dev.Addrs {
+		if addr.Addr.IsLoopback() || addr.Addr.IsLinkLocalUnicast() {
+			continue
+		}
+		if family.Afi == types.AfiIPv4 && !addr.Addr.Is4() {
+			continue
+		}
+		if family.Afi == types.AfiIPv6 && !addr.Addr.Is6() {
+			continue
+		}
+		res = append(res, netip.PrefixFrom(addr.Addr, addr.Addr.BitLen()))
```

Comment:
Nit: This logic may introduce duplicated advertisement, for example, when there's `10.0.0.0/24` and `10.0.0.0/32` assigned to the interface. Maybe better to deduplicate addresses before returning.